tests/vlan/test_vlan.py

```diff
@@ -345,3 +359,112 @@ def test_vlan_tc3_send_invalid_vid(ptfadapter, vlan_ports_list):
         testutils.send(ptfadapter, src_port, invalid_tagged_pkt)
         logger.info("Check on " + str(dst_ports) + "...")
         testutils.verify_no_packet_any(ptfadapter, masked_invalid_tagged_pkt, dst_ports)
+
+
+@pytest.mark.bsl
+def test_vlan_tc4_tagged_non_broadcast(ptfadapter, vlan_ports_list, duthost):
+    """
+    Test case #4
+    Send packets w/ src and dst specified over tagged ports in vlan
+    Verify that bidirectional communication between two tagged ports work
+    """
+    duthost.command('sonic-clear fdb all')  # clear mac learning
+
+    vlan_ids = vlan_ports_list[0]['permit_vlanid'].keys()
+    tagged_test_vlan = vlan_ids[0]
+    untagged_test_vlan = vlan_ids[1]
+
+    ports_for_test = []
+
+    for vlan_port in vlan_ports_list:
+        if vlan_port['pvid'] != tagged_test_vlan:
+            ports_for_test.append(vlan_port['port_index'][0])
+
+    # take two tagged ports for test
+    src_port = ports_for_test[0]
+    dst_port = ports_for_test[-1]
```

Comment:
Does this assume that port 0 and port (len-1) will be in tagged mode on the DUT?

Reply:
Both ports will be tagged ports, yes. The ports_for_test are a list of just the tagged ports on the vlan